src/main/java/uk/gov/justice/hmpps/prison/service/FinanceService.java

```diff
@@ -41,11 +48,21 @@
     private final AccountCodeRepository accountCodeRepository;
     private final OffenderSubAccountRepository offenderSubAccountRepository;
     private final OffenderTrustAccountRepository offenderTrustAccountRepository;
+    private final OffenderDamageObligationService offenderDamageObligationService;
 
     @VerifyBookingAccess(overrideRoles = {"SYSTEM_USER", "GLOBAL_SEARCH"})
     public Account getBalances(final Long bookingId) {
+
+        var damageObligationBalance = bookingRepository.getLatestBookingByBookingId(bookingId)
+            .map(OffenderSummary::getOffenderNo)
```

Comment:
There is a fair amount of indirection going on here and it makes the code hard to comprehend. Would it be worth breaking it out a bit?  It could possibly be simplified without the need for all those **private static final Function<>**

Reply:
happy to simplify, I think setDamageObligation shows clear intentions but I've refactored it to 
```
private static final Function<Pair<Account, BigDecimal>, Account> setDamageObligation = (pair) -> {
        val account = pair.getFirst();
        val damageObligationAmount = pair.getSecond();
        account.setDamageObligations(damageObligationAmount);
        return account;
    };
```
Does this look better?

I'm trying to avoid multi-line code blocks that are less expressive and make the reader put more effort to understand the logic.

Reply:
I think sometimes multi line code blocks can help. Especially to help keep logical steps separate. This hasn't been tested and you could probably making it even neater but, something like the following means the reader doesn't have to chase around to understand what's going on:

`final var booking = bookingRepository.getLatestBookingByBookingId(bookingId).orElseThrow(new EntityNotFoundException("Booking not found for x"));

        final var damageObligationBalance =
            offenderDamageObligationService.getDamageObligations(booking.getOffenderNo(), "")
                .stream()
                .map(OffenderDamageObligationModel::getAmountToPay)
                .reduce(BigDecimal.ZERO, BigDecimal::add);

        return financeRepository.getBalances(bookingId, booking.getAgencyLocationId())
            .toBuilder()
            .damageObligations(damageObligationBalance)
            .build();`

Reply:
done